### PR TITLE
fix: bug on 0 supply pricing

### PR DESCRIPTION
**Pull request by fubuloubu**

### CODE REVIEW

**Comment by fubuloubu** [comment]
compare to #4

**Comment by transmissions11** [comment]
can incorporate this into #4 no?

**Comment by fubuloubu** [comment]
> can incorporate this into #4 no?

Yep, either way is fine

Really need to set up tests here lol

File: contracts/Vault.vy [hunk 1]
```diff
@@ -99,7 +99,11 @@ def totalAssets() -> uint256:
 @view
 @internal
 def _calculateAssets(shareAmount: uint256) -> uint256:
-    return shareAmount * self.asset.balanceOf(self) / self.totalSupply
+    totalSupply: uint256 = self.totalSupply
+    if totalSupply == 0:
+        return 0
+
+    return shareAmount * self.asset.balanceOf(self) / totalSupply
```
Comment (fubuloubu) [inline contracts/Vault.vy:106]:
```suggestion
    # NOTE: `shareAmount = 0` is extremely rare case, not optimizing for it
    # NOTE: `totalAssets = 0` is extremely rare case, not optimizing for it
    return shareAmount * self.asset.balanceOf(self) / totalSupply
```

File: contracts/Vault.vy [hunk 2]
```diff
@@ -117,7 +121,11 @@ def assetsOf(owner: address) -> uint256:
 @view
 @internal
 def _calculateShares(assetAmount: uint256) -> uint256:
-    return assetAmount * self.totalSupply / self.asset.balanceOf(self)
+    totalAssets: uint256 = self.asset.balanceOf(self)
+    if totalAssets == 0:
+        return assetAmount
+
+    return assetAmount * self.totalSupply / totalAssets
```
Comment (fubuloubu) [inline contracts/Vault.vy:128]:
```suggestion
    # NOTE: `assetAmount = 0` is extremely rare case, not optimizing for it
    # NOTE: `totalSupply = 0` is extremely rare case, not optimizing for it
    return assetAmount * self.totalSupply / totalAssets
```

**Pull request merged**